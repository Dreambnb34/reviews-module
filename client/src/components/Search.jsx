import React, {Component} from 'react';

const searchStyle = {
  height: '1em',
  width: '1em',
  display: 'block',
  fill: 'currentColor',
};

const otherStyle = {
  visibility: 'visible',
  height: '32px',
};

class Search extends Component {
  constructor(props) {
    super(props);
    this.state = {value: ''};

    this.handleChange = this.handleChange.bind(this);
    this.handleKeyPress = this.handleKeyPress.bind(this);
  }

  handleChange(e) {
    e.preventDefault();
    this.setState({value: e.target.value});
  }

  handleKeyPress(e) {
    if (e.key === 'Enter') {
      e.preventDefault();
      console.log('Here!!!!!!!');
      // From props up top, start search
      this.props.onPageSearch(this.state.value);
      this.setState({value: ''});
    }
  }

  render() {
    return (
      <div className="search-container">
        <form>
          <input
            type="text"
            value={this.state.value}
            onChange={this.handleChange}
            className="search-form"
            placeholder="Search reviews"
            onKeyDown={this.handleKeyPress}
          />
        </form>
      </div>
    );
  }
}

export default Search;
