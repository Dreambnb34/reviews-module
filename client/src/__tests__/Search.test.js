import React from 'react';
import {render, fireEvent} from 'react-testing-library';
import Search from '../components/Search';

describe('Search Component', () => {
  it('renders the search component', () => {
    const {container} = render(<Search />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it('should intiate search when the "Enter" key is pressed', () => {
    const mockFn = jest.fn();
    const {container} = render(<Search onPageSearch={mockFn} />);
    fireEvent.keyDown(container.getElementsByTagName('input')['0'], {
      key: 'Enter',
      keyCode: 13,
      which: 13,
    });
    expect(mockFn.mock.calls.length).toBe(1);
  });

  it('should intiate search with the correct value in the input field', () => {
    const mockFn = jest.fn();
    const {getByPlaceholderText} = render(<Search onPageSearch={mockFn} />);
    const input = getByPlaceholderText('Search reviews');
    fireEvent.change(input, {
      target: {value: 'Test'},
    });

    expect(input.value).toBe('Test');
  });
});
