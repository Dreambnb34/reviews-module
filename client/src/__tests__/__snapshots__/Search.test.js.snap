// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search Component renders the search component 1`] = `
<div
  class="search-container"
>
  <form>
    <input
      class="search-form"
      placeholder="Search reviews"
      type="text"
      value=""
    />
  </form>
</div>
`;
